### PR TITLE
Improve nix/purescript workflow

### PR DESCRIPTION
**Pull request by gilligan**
This PR addresses some discrepancies between the usual developer workflow for the frontend clients and the nix build - more specifically the handling of `web-common`. More specifically this PR:

- symlinks `web-common` to plutus,marlowe and scb-client
- updates references to `../web-common`
- updates the webpack configuration so npm dependencies can be found when loaded from `web-common`

Additionally this PR introduces scripts to execute the purescript code generation:
- `plutus-playground-generate-purs`
- `marlowe-playground-generate-purs`
- `plutus-scb-generate-purs`

When executed they will write the respective code to `./generated` (after removing any potentially existing previous `generated` directory).

**Note**: nix builds and the development cycle workflow both seem to be working with this. I also asked @merivale to verify this. 

-----

Pre-submit checklist:
- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested
- If you updated any cabal files or added Haskell packages:
    - [ ] `$(nix-build default.nix -A plutus.updateMaterialized)` to update the materialized Nix files
   - [ ] Update `hie-*.yaml` files if needed
- If you changed any Haskell files:
   - [ ] `$(nix-shell shell.nix --run fix-stylish-haskell)` to fix any formatting issues
- If you changed any Purescript files:
   - [ ] `$(nix-shell shell.nix --run fix-purty)` to fix any formatting issues

Pre-merge checklist:
- [x] Someone approved it
- [x] Commits have useful messages
- [x] Review clarifications made it into the code
- [x] History is moderately tidy; or going to squash-merge

### CODE REVIEW

File: marlowe-playground-client/web-common
Comment (shmish111) [inline marlowe-playground-client/web-common:1]: what is this file?

**Review by shmish111 — APPROVED** [review]
LGTM

File: marlowe-playground-client/web-common
Comment (michaelpj) [reply marlowe-playground-client/web-common:1]: This is how github shows a checked-in symlink.

File: marlowe-playground-client/package.json
Comment (merivale) [inline marlowe-playground-client/package.json:10]: `'./web-common/**/*.purs'` could just be `'web-common/**/*.purs'`

File: marlowe-playground-client/spago.dhall
Comment (merivale) [inline marlowe-playground-client/spago.dhall:40]: `"./web-common/**/*.purs"` could just be `"web-common/**/*.purs"`

File: marlowe-playground-client/webpack.config.js
Comment (merivale) [inline marlowe-playground-client/webpack.config.js:74]: `'./web-common/**/*.purs'` could just be `'web-common/**/*.purs'`

File: marlowe-playground-client/webpack.config.js
Comment (merivale) [inline marlowe-playground-client/webpack.config.js:110]: Just curious why this is necessary? Is it to do with the symlink?

File: marlowe-playground-client/webpack.test.config.js
Comment (merivale) [inline marlowe-playground-client/webpack.test.config.js:43]: As with the others.

File: nix/lib/purescript.nix
Comment (merivale) [inline nix/lib/purescript.nix:42]: Here too.

File: plutus-playground-client/package.json
Comment (merivale) [inline plutus-playground-client/package.json:10]: And again.

File: plutus-playground-client/spago.dhall
Comment (merivale) [inline plutus-playground-client/spago.dhall:36]: Another one.

**Review by merivale — COMMENTED** [review]
Looking good. But I think we can do a bit more before merging. Sorting out the npm scripts (as we've talked about in slack). Also we could remove some noise from the `.gitignore` files (noise that I created while you were on holiday 😁 ).

File: marlowe-playground-client/package.json
Comment (gilligan) [reply marlowe-playground-client/package.json:10]: you are right, that is equivalent - is there any benefit that the one has over the other though?

Comment (gilligan) [reply marlowe-playground-client/package.json:10]: never mind - i'll just change it everywhere.

File: marlowe-playground-client/webpack.config.js
Comment (gilligan) [reply marlowe-playground-client/webpack.config.js:110]: Maybe i should add a comment about this in the code:

This adds the absolute path of `node_modules` in the respective client directory (plutus-playground or marlowe etc) to the search path for modules. Without it, node will fail to find some npm dependency that it tries to load from within the `web-common` directory.

Comment (gilligan) [reply marlowe-playground-client/webpack.config.js:113]: I added comments in https://github.com/input-output-hk/plutus/pull/2608/commits/977a20241e240bc3fe0cd6a2c0e943a4d78a095a

**Comment by gilligan** [comment]
> Looking good. But I think we can do a bit more before merging. Sorting out the npm scripts (as we've talked about in slack). Also we could remove some noise from the `.gitignore` files (noise that I created while you were on holiday  ).

I agree that we can improve things further and i have some local changes already but i would like to not make this PR any bigger. It does change the build behavior already so i would like to implement the other parts in a separate PR that i can create shortly after this one.

**Comment by merivale** [comment]
Sure. Your call.

**Review by merivale — APPROVED** [review]

File: shell.nix
Comment (michaelpj) [inline shell.nix:62]: This makes `shell.nix` depend on `plutus-playground-server` and hence basically all of our Haskell packages transitively, which makes it much slower and causes lorri to thrash (since the shell derivation will change any time you change any source file).

I recommend we make this an attribute and tell people to run it with a `$(nix-build ...)` invocation. That's what we do with `updateMaterialized` for the same reason.

**Pull request merged**
